enhancements/kube-apiserver/bound-sa-tokens.md

```diff
@@ -0,0 +1,175 @@
+    in kube > 1.12, but are only configured by the apiserver if the following options are
+    provided:
+    - service-account-signing-key-file
+      - Operator should set this to the path of the private key it manages
+    - service-account-issuer
+      - Operator should default this to `auth.openshift.io` and it should be possible to
+        override it. When it is overriden, tokens from the previous issuer will no longer
+        validate and it may be necessary to restart affected pods.
+    - api-audiences
+      - Operator should set this to the issuer.
+      - Bound tokens submitted to the apiserver must specify one or more of the audiences
+        configured here. If a request for a bound token does not specify an audience, the
+        audience will be defaulted to the issuer.
+  - Enabling bound tokens will satisfy goals #1 and #2.
+
+- The apiserver operator should manage a keypair used to sign and verify bound tokens
+  - An rsa keypair will be used. The trust model for digital signatures (via jwt) is
+    different than for securing web communciation, and no explicit provision for expiry
+    or revocation should be necessary.
+  - The keypair needs to be distinct from the keypair used for legacy service account
+    tokens. Reusing the legacy token keypair would prevent token validation since tokens
+    are first validated by key and then by content. If the key were to validate, the
+    subsequent content check would fail due to differences in the content of legacy and
+    bound tokens.
+  - The keypair should be written to a secret in the `openshift-kube-apiserver` namespace
+  - The public key should be added to (rather than replaced in) a configmap in the
+    `openshift-kube-apiserver` namespace.
+    - Public keys will be added to the configmap with keys of the form
+      `bound-service-account-xxx.pub`, where `xxx` is incremented for uniqueness.
+    - This configmap can be used to source the public keys needed by 3rd party components
+      to verify bound tokens issued by the apiserver, satisfying goal #3.
+    - The path of the mounted configmap should be included in
+      `KubeAPIServerConfig.ServiceAccountPublicKeyFiles` to configure the apiserver. The
+      path will be automatically translated into the list of filenames in that path
+      during conversion from openshift configuration to apiserver configuration.
+    - If it is necessary to invalidate previous tokens, deletion of the configmap will
+      ensure recreation with only the current public key.
+  - In the event that the keypair secret is deleted, a new keypair will be generated but
+    tokens signed by the previous private key will continue to validate since the
+    corresponding public key will still be used to validate bound tokens.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Bound tokens are not currently required until bootstrap is complete, so it should be
```

Comment:
cc @bergerhoffer, this needs to go into docs as a warning

Reply:
Thanks @sttts I'll make a note of this.